app/services/container_dashboard_service.rb

```diff
@@ -20,8 +20,8 @@ def all_data
   end
 
   def status
-    if @ems.present? && @ems.kind_of?(ManageIQ::Providers::Openshift::ContainerManager)
-      routes_count = @ems.container_routes.count
+    if @ems.present?
+      routes_count = @ems.respond_to?(:container_routes) ? @ems.container_routes.count : 0 # ems might not have routes
```

Comment:
> perhaps you are setting routes as zero always even for k8s?

always setting routes to zero for k8s.
ems_k8s doesnt have ```:has_many :container_routes``` so it would cause an exception. we could go with something like 
```Ruby
routes_count = if @ems.present?
  @ems.try(:container_routes).try(:count) || 0
else
  ContainerRoute.count
end
```